### PR TITLE
Fix bug tls renegociation problem in Notification channel (webhook) #…

### PR DESCRIPTION
**Pull request by fredbcode**
Fix for bug #14800

### CODE REVIEW

File: pkg/services/notifications/webhook.go [hunk 1]
```diff
@@ -26,6 +26,10 @@ type Webhook struct {
 }
 
 var netTransport = &http.Transport{
+	TLSClientConfig: &tls.Config{
+		InsecureSkipVerify: true,
```
Comment (torkelo) [inline pkg/services/notifications/webhook.go]: this seems like a reduction in security that not everyone would be ok with, is it required for this fix?

Comment (fredbcode) [reply pkg/services/notifications/webhook.go]: Yes, I based my approach on the work of BrandonArp here https://github.com/grafana/grafana/commit/cffbcb504d884022f41fc48a6099d15018e01863
The request is from grafana to a distant server not the reverse

Comment (bergquist) [reply pkg/services/notifications/webhook.go]: Hm. I thought we removed all those some time ago. `InsureSkipVerify` should not be true by default.

Comment (fredbcode) [reply pkg/services/notifications/webhook.go]: Ok, I thought the the both was related - InsecureSkipVerify: true, and Renegotiation -, I can push a merge without this entry ?

Comment (bergquist) [reply pkg/services/notifications/webhook.go]: Yes we did the commit you refer to is very old. 
https://github.com/grafana/grafana/blob/master/pkg/models/datasource_cache.go#L58

Comment (bergquist) [reply pkg/services/notifications/webhook.go]: please do :)

Comment (fredbcode) [reply pkg/services/notifications/webhook.go]: Yes I see, unfortunately tlsSkipVerify is a value from datasource, https://github.com/grafana/grafana/pull/9377 not global
So I guess the best think to do make a checkbox like as datasource (but beyond my knowledge of go)

InsureSkipVerify is not related with Renegotiation: tls.RenegotiateFreelyAsClient,

So I can push a request with only this entry ? Anyway tls.RenegotiateFreelyAsClient is a default value in your code now

Comment (bergquist) [reply pkg/services/notifications/webhook.go]: only pushing `Renegotiation: tls.RenegotiateFreelyAsClient,` makes total sense.

Comment (fredbcode) [reply pkg/services/notifications/webhook.go]: And add a checkbox to your todo list for notification tab :)

Comment (fredbcode) [reply pkg/services/notifications/webhook.go]: Ok pushed

Comment (bergquist) [reply pkg/services/notifications/webhook.go]: you need to import "crypto/tls" in the top of the file

Comment (fredbcode) [reply pkg/services/notifications/webhook.go]: ok done

Comment (bergquist) [reply pkg/services/notifications/webhook.go]: you need to run `gofmt -s -w ./pkg` in the root of the grafana folder to get the correct formating.

Comment (fredbcode) [reply pkg/services/notifications/webhook.go]: Ok, there are also 3 others files "fixed"

**Comment by bergquist** [comment]
Not sure I understand the last commit `09f8114`

**Comment by fredbcode** [comment]
It's just a result of gofmt -s -w ./pkg

**Comment by bergquist** [comment]
weird. Cause that does not look formated :) and is rejected in CI due to formating. 
What version of golang are you using?

**Comment by fredbcode** [comment]
Debian strech:

dpkg -l |grep golint
ii  golint                               0.0+git20161013.3390df4-1                      amd64        Linter for Go source code

**Comment by fredbcode** [comment]
I'm not a go expert , I can't tell, but it's a separate pull maybe you can just merge the rest ?

**Comment by bergquist** [comment]
Sorry for the delay. 
I've fixed and merge this. 

Since I could not push to your branch I merged this locally
https://github.com/grafana/grafana/commit/110ffb69dee696dca305167d0167c4c78462fde8

**Pull request closed without merging**